### PR TITLE
Saya Proving

### LINKED ISSUE #1653: [saya] integrate the prover
In this issue, we track the integration of the prover into Saya.
This work must be done for now into the `dev/katana` branch.

The prover trait must be present into this module https://github.com/dojoengine/dojo/blob/027976df1f2e309f8c075871d76925f4bed31392/crates/saya/core/src/prover/mod.rs, where any additional info can be added to support multi prover backend.

Then, the first implementation should focus on generating a proof that can be then verified by the cairo verifier.

### PR DESCRIPTION
**Pull request by neotheprogramist**
closes DOJ-255
closes DOJ-256
closes #1653 #1654

### CODE REVIEW

**Comment by glihm** [comment]
Will be rebased once `/dev/katana` is merged into main. :+1:

File: crates/saya/core/src/verifier/mod.rs [hunk 1]
```diff
@@ -7,9 +7,26 @@
 //! an interface to query the on-chain verifier, but also
 //! submitting facts and proofs.
 
+use crate::prover::parse_proof;
+mod starknet;
+
 /// Supported verifiers.
 #[derive(Debug)]
 pub enum VerifierIdentifier {
+    HerodotusStarknetSepoia,
```
Comment (apoorvsadana) [inline crates/saya/core/src/verifier/mod.rs:16]:
```suggestion
    HerodotusStarknetSepolia,
```

:)

**Comment by glihm** [comment]
@matzayonc this was closed due to the rebase of Katana.
If you prefer having it open again in this one I will change the target branch. If you prefer doing it in a new PR, let me know. :+1:

**Comment by matzayonc** [comment]
> @matzayonc this was closed due to the rebase of Katana. If you prefer having it open again in this one I will change the target branch. If you prefer doing it in a new PR, let me know. 👍

Sure, I think we can keep it. Should it just be to the `main` now?

**Comment by matzayonc** [comment]
> @matzayonc this was closed due to the rebase of Katana. If you prefer having it open again in this one I will change the target branch. If you prefer doing it in a new PR, let me know. 👍

It way much easier to create a new branch and cherry-pick, it is now `feat/saya-prove-verify`

**Pull request closed without merging**